### PR TITLE
Fix: Progress bar continue to update after footer

### LINKED ISSUE #4023: Status bar logged to console after completion message
**Do you want to request a *feature* or report a *bug*?**
Bug

**What is the current behavior?**
Status bar logged to console after completion message is printed.

**If the current behavior is a bug, please provide the steps to reproduce.**
Use latest yarn from master for this:
```
yarn add express
```

You'll see items being logged after the completion message "Done in Xs" is printed. I think it's related to commit 61e402b where `process.exit()` was changed out for `process.exitCode`;

screenshot

**What is the expected behavior?**
Completion message is the final thing printed.

**Please mention your node.js, yarn and operating system version.**
Node.js v6.11.0 / Yarn latest commit

### PR DESCRIPTION
**Pull request by BYK**
**Summary**

Fixes #4023. There was nothing preventing the console reporter from
having more than one progress bars, or keep updating and rendering
its progress bar after it was finished or the reporter was "done".

This patch stores the active progress bar, and stops it before
`footer` is printed out. Also makes sure the progress bar itself
ignores any updates once it stops.

**Test plan**

One, weak, existing test with updated snapshot. Manual steps:

- Run `yarn install express`
- Observe that your console is properly cleared after `yarn`
  finishes and the last thing you see is the "Done in X.YZs."
  message and not a corrupted progress bar.

### CODE REVIEW

File: src/reporters/console/progress-bar.js [hunk 1]
```diff
@@ -20,28 +21,44 @@ export default class ProgressBar {
   chars: [string, string];
   delay: number;
   id: ?number;
+  _callback: ?Function;
```
Comment (cpojer) [inline src/reporters/console/progress-bar.js]: Can you properly type this callback function?

File: src/reporters/console/progress-bar.js [hunk 2]
```diff
@@ -20,28 +21,44 @@ export default class ProgressBar {
...
     }
+  }
 
-    // progress complete
-    if (this.curr >= this.total) {
+  cancelTick() {
+    if (!this.id) {
       clearTimeout(this.id);
-      clearLine(this.stdout);
+      this.id = null;
```
Comment (mjesun) [inline src/reporters/console/progress-bar.js]: If `!this.id` then the timeout is cleared?

Comment (BYK) [inline src/reporters/console/progress-bar.js]: Yup

File: src/reporters/console/progress-bar.js (hunk 1 above)
Comment (BYK) [inline src/reporters/console/progress-bar.js]: (y)

File: src/reporters/console/progress-bar.js (hunk 2 above)
Comment (mjesun) [inline src/reporters/console/progress-bar.js]: Nope

**Pull request merged**